roles/ceph-mon/tasks/deploy_monitors.yml

```diff
@@ -32,14 +36,30 @@
     group: "{{ dir_group }}"
     mode: "{{ dir_mode }}"
 
-- name: ceph monitor mkfs (for or after infernalis release)
+- name: ceph monitor mkfs with keyring (for or after infernalis release)
   command: ceph-mon --setuser ceph --setgroup ceph --mkfs -i {{ ansible_hostname }} --fsid {{ fsid }} --keyring /var/lib/ceph/tmp/keyring.mon.{{ ansible_hostname }}
   args:
     creates: /var/lib/ceph/mon/ceph-{{ ansible_hostname }}/keyring
-  when: is_ceph_infernalis
+  when:
+    cephx and
+    is_ceph_infernalis
 
-- name: ceph monitor mkfs (before infernalis release)
+- name: ceph monitor mkfs without keyring (for or after infernalis release)
+  command: ceph-mon --setuser ceph --setgroup ceph --mkfs -i {{ ansible_hostname }} --fsid {{ fsid }}
```

Comment:
well if we remove the create, we don't ensure the idempotency anymore so we need to find a file that will get created by the monitor. Something like a lvldb file should do it.